iree/test/e2e/tosa_ops/pad.mlir

```diff
@@ -0,0 +1,8 @@
+
+func @pad_test() attributes { iree.module.export } {
+    %0 = "tosa.const"() { value = dense<1> : tensor<1x1x1xi32> } : ()  -> (tensor<1x1x1xi32>)
+    %1 = "tosa.const"() { value = dense<[[0, 1], [1, 0], [0, 0]]> : tensor<3x2xi32> } : ()  -> (tensor<3x2xi32>)
+    %result = "tosa.pad"(%0, %1)  : (tensor<1x1x1xi32>, tensor<3x2xi32>)  -> (tensor<2x2x1xi32>)
+    check.expect_eq_const(%result, dense<[[[0], [1]], [[0], [0]]]> : tensor<2x2x1xi32>) : tensor<2x2x1xi32>
```

Comment:
o_O seems to match the spec, but wow the spec is confusing... 

After reading through these:
https://mlir.llvm.org/docs/Dialects/TOSA/#tosapad-mlirtosapadop
https://developer.mlplatform.org/w/tosa/
and referencing this test, I'm still not sure where `2x2x1` is coming from, or why these values are expected...

Could you use values other than 0 and 1 to more clearly show what is happening?

Reply:
I added two more tests and used the internal value as 42 to be more clear. I can not change what the padding value is, so you are stuck with the 0s. Not sure we are going to get clearer with padding.